tamproxy/comm/packet_controller.py

```diff
@@ -180,36 +185,40 @@ def update_timeout(self, time_sent):
         self.timeout = self.SRTT_GAIN*self.rtt_smoothed + self.RTTDEV_GAIN*self.rtt_deviation
 
     def stop(self):
+        logger.info('stop requested')
         self._stop.set()
 
     def run(self):
+        # keyboard interrupt is handled by the main process
+        signal.signal(signal.SIGINT, signal.SIG_IGN)
+
         i = self.SERIAL_RETRIES
         while i >= 0:
             try:
+                logger.info('Connecting to the teensy')
                 self.connect()
+                logger.info('Connected!')
                 i = self.SERIAL_RETRIES
-                while not self._stop.is_set():
+                while True:
                     self.slide_window()
                     self.receive()
-                return
+                    if self._stop.is_set() and not self.pipe_inside.poll():
```

Comment:
On the whole, great change! I don't think pipe_inside.poll() is sufficient to check, though. We should also check len(en_route) == 0, since our packets may not get through immediately.